jrqs-app/src/java/ua/dp/isd/jrqs/RequirementsService.java

```diff
@@ -27,22 +27,40 @@
 package ua.dp.isd.jrqs;
 
 import java.util.List;
+import ua.dp.isd.jrqs.ESearchBy.SearchBy;
 
 public class RequirementsService {
-    
-    public List<Requirement> getRequirementsByRowId(String RequirementRowIds) {
-        return new RequirementsDAO().getRequirementsByRowIds(RequirementRowIds);
-    }
-    
-    public List<Requirement> getRequirementsByRequirementNumbers(String RequirementNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByRequirementNumbers(RequirementNumbers, LimitBySource);
+
+    private final String rawParamsSearchBy;
+    private final String rawParamsUserInput;
+    private final String rawParamsLimitBySource;
+
+    public RequirementsService(String by, String value, String only) {
+        rawParamsSearchBy = by;
+        rawParamsUserInput = value;
+        rawParamsLimitBySource = only;
     }
-    
-    public List<Requirement> getRequirementsByTmsTaskNumbers(String TmsTaskNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByTmsTaskNumbers(TmsTaskNumbers, LimitBySource);
+
+    public List<Requirement> getRequirements() {
+        if (rawParamsSearchBy == null || rawParamsSearchBy.equals("")) {
+            throw new IllegalArgumentException("Invalid request: by parameter is missing in the request.");
+        }
+        SearchBy by = detectSearchBy(rawParamsSearchBy);
+        RequirementsDAO dao = new RequirementsDAO();
+        return dao.getRequirements(by, rawParamsUserInput, rawParamsLimitBySource);
     }
     
-    public List<Requirement> getRequirementsByTextPhrases(String TextPhrases, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByTextPhrases(TextPhrases, LimitBySource);
+    private SearchBy detectSearchBy(String by) {

```

Comment:
It looks like this class is business logic. In this case, you should not pass there any raw user data. It's better to convert raw strings to enum's somewhere closer to UI and use in business logic only business objects. Everything related to SQL HAVE to be abstracted as well.